### ISSUE TITLE
feat: pyspark create_table enable delta format

### ISSUE THREAD

**Issue by mark-druffel** [ibis-project/ibis#9032]
### Is your feature request related to a problem?

I'm working on databricks with unity catalog (think it's just hive metastore with extra features). I'm trying to write managed tables to unity catalog with ibis which requires me to save with delta lake instead of parquet. 

In pyspark, I can write a managed table like so:
```
from pyspark.sql import SparkSession
spark.catalog.setCurrentCatalog("comms_media_dev")
spark.catalog.setCurrentDatabase("dart_extensions")
df = spark.read.parquet("abfss://my_parquet")
df = df.select("CAMP_START_DATE", "CAMP_END_DATE", "KPM_PROJECT_ID")
df.write.saveAsTable(name = "raw_camp_info", mode = "overwrite", format = "delta")
spark.sql("DROP TABLE raw_camp_info")
```

I wanted to do the same thing with ibis create_table, but I get an error. I'm not sure if maybe I'm doing something wrong or I need to upgrade pyspark. I'm on pyspark 3.2.2 because of some other dependencies, but c
```
import ibis
from ibis import _
spark = SparkSession.builder.getOrCreate()
ispark = ibis.pyspark.connect(session = spark)
ispark._session.catalog.setCurrentCatalog("comms_media_dev")
ispark._session.catalog.setCurrentDatabase("dart_extensions")
df = ispark.read_parquet(source = "abfss://my_parquet")
df = df.select(_.CAMP_START_DATE, _.CAMP_END_DATE, _.KPM_PROJECT_ID)
ispark.create_table(name = "raw_camp_info", obj = df, overwrite = True, format="delta")
```
> ValueError: Invalid format: 'DELTA'

> File <command-1558648439768505>, line 10
>       8 df = ispark.read_parquet(source = "abfss://my_parquet")
>       9 df = df.select(_.CAMP_START_DATE, _.CAMP_END_DATE, _.KPM_PROJECT_ID)
> ---> 10 ispark.create_table(name = "raw_camp_info", obj = df, overwrite = True, format="delta")
> File /local_disk0/.ephemeral_nfs/envs/pythonEnv-bdae000a-2b7c-424c-b604-b22e32576b08/lib/python3.10/site-packages/ibis/backends/pyspark/__init__.py:421, in Backend.create_table(self, name, obj, schema, database, temp, overwrite, format)
>     418     ast = self.compiler.to_ast(obj)
>     419     select = ast.queries[0]
> --> 421     statement = ddl.CTAS(
>     422         name,
>     423         select,
>     424         database=database,
>     425         can_exist=overwrite,
>     426         format=format,
>     427     )
>     428 else:
>     429     statement = ddl.CreateTableWithSchema(
>     430         name,
>     431         schema,
>    (...)
>     434         can_exist=overwrite,
>     435     )
> File /local_disk0/.ephemeral_nfs/envs/pythonEnv-bdae000a-2b7c-424c-b604-b22e32576b08/lib/python3.10/site-packages/ibis/backends/pyspark/ddl.py:93, in CTAS.__init__(self, table_name, select, database, format, can_exist)
>      85 def __init__(
>      86     self,
>      87     table_name,
>    (...)
>      91     can_exist=False,
>      92 ):
> ---> 93     super().__init__(
>      94         table_name,
>      95         select,
>      96         database=database,
>      97         format=format,
>      98         can_exist=can_exist,
>      99     )
>     100     self.select = select
> File /local_disk0/.ephemeral_nfs/envs/pythonEnv-bdae000a-2b7c-424c-b604-b22e32576b08/lib/python3.10/site-packages/ibis/backends/base/sql/ddl.py:179, in CTAS.__init__(self, table_name, select, database, external, format, can_exist, path, partition)
>     168 def __init__(
>     169     self,
>     170     table_name,
>    (...)
>     177     partition=None,
>     178 ):
> --> 179     super().__init__(
>     180         table_name,
>     181         database=database,
>     182         external=external,
>     183         format=format,
>     184         can_exist=can_exist,
>     185         path=path,
>     186         partition=partition,
>     187     )
>     188     self.select = select
> File /local_disk0/.ephemeral_nfs/envs/pythonEnv-bdae000a-2b7c-424c-b604-b22e32576b08/lib/python3.10/site-packages/ibis/backends/base/sql/ddl.py:135, in CreateTable.__init__(self, table_name, database, external, format, can_exist, partition, path, tbl_properties)
>     133 self.external = external
>     134 self.can_exist = can_exist
> --> 135 self.format = _sanitize_format(format)
>     136 self.tbl_properties = tbl_properties
> File /local_disk0/.ephemeral_nfs/envs/pythonEnv-bdae000a-2b7c-424c-b604-b22e32576b08/lib/python3.10/site-packages/ibis/backends/base/sql/ddl.py:22, in _sanitize_format(format)
>      20 format = _format_aliases.get(format, format)
>      21 if format not in ("PARQUET", "AVRO", "TEXTFILE"):
> ---> 22     raise ValueError(f"Invalid format: {format!r}")
>      24 return format
> 

### What is the motivation behind your request?

I am able to work around the issue using views and the spark session:
```
import ibis
from ibis import _
spark = SparkSession.builder.getOrCreate()
ispark = ibis.pyspark.connect(session = spark)
ispark._session.catalog.setCurrentCatalog("comms_media_dev")
ispark._session.catalog.setCurrentDatabase("dart_extensions")
idf = ispark.read_parquet(source = "abfss://my_parquet")
idf = df.select(_.CAMP_START_DATE, _.CAMP_END_DATE, _.KPM_PROJECT_ID)
ispark.create_view(name = "raw_camp_info", obj = idf, overwrite = True)
sdf = ispark._session.sql("SELECT * FROM raw_camp_info")
sdf.write.saveAsTable(name = "raw_camp_info", mode = "overwrite", format = "delta")
```

The above approach will work, but it's a lot clunkier... 

### Describe the solution you'd like

I would like to be able to use the delta format with `create_table`.
ispark.create_table(name = "raw_camp_info", obj = df, overwrite = True, format="delta")

### What version of ibis are you running?

8.0.0

### What backend(s) are you using, if any?

pyspark, tested with 3.2.2 & 3.5.0

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by gforsyth** [comment]
Hey @mark-druffel -- thanks for the report!  This should be fixed on `main` if you want to try out a pre-release.  We're no longer sanitizing the inputs to `format` so if the underlying spark version supports writing to `delta`, it should "just work" since `create_table` is calling `saveAsTable` under the hood.

**Comment by mark-druffel** [comment]
@gforsyth Thanks, the pre-release works great! I do have one other question though 😬 Let me know if this question deserves it's own issue. 

## TLDR
I'm wondering if it's intended that the `database` argument in `create_table` works different than the one in `drop_table`? `create_table` only accepts a `str` and `drop_table` accepts a `tuple`. 

If I set the catalog and database via pyspark, `create_table` works as excepted, but I can't figure out a way to set the catalog and database without accessing the pyspark session directly:
```
ispark._session.catalog.setCurrentCatalog("comms_media_dev")
ispark._session.catalog.setCurrentDatabase("dart_extensions")
ispark.create_table(name = "raw_camp_info", obj = df, overwrite = True, format="delta")
```
I can drop a table without accessing the pyspark session:
```
from pyspark.sql import SparkSession
import ibis
spark = SparkSession.builder.getOrCreate()
ispark = ibis.pyspark.connect(session = spark)
ispark.drop_table(name = "raw_camp_info", database=tuple(["comms_media_dev", "dart_extensions"]))
```

## Additional Details
To drop my table I can just specify the catalog and database in my call:
```
from pyspark.sql import SparkSession
import ibis
spark = SparkSession.builder.getOrCreate()
ispark = ibis.pyspark.connect(session = spark)
ispark.drop_table(name = "raw_camp_info", database=tuple(["comms_media_dev", "dart_extensions"]))
```

Trying the same approach with `create_table` if fails:
```
ispark.create_table(name = "raw_camp_info", obj = df, overwrite = True, format="delta", database=tuple(["comms_media_dev", "dart_extensions"]))
```
> py4j.Py4JException: Method setCurrentDatabase([class java.util.ArrayList]) does not exist

I also tried with dot separator:
```
ispark.create_table(name = "raw_camp_info", obj = df, overwrite = True, format="delta", database="comms_media_dev.dart_extensions")
```
> AnalysisException: too many name parts for schema name: comms_media_dev.dart_extensions

I then tried to set the catalog and provide the database name and got a permissions error. Looking through the error, it looks like create_table didn't pass the `database` argument because the database was set to default (i.e. `comms_media_dev.default`:
```
ispark._session.catalog.setCurrentCatalog("comms_media_dev")
ispark.create_table(name = "raw_camp_info", obj = df, overwrite = True, format="delta", database="dart_extensions")
```

> com.databricks.sql.managedcatalog.acl.UnauthorizedAccessException: PERMISSION_DENIED: User does not have USE SCHEMA on Schema 'comms_media_dev.default'.
> ---------------------------------------------------------------------------
> Py4JJavaError                             Traceback (most recent call last)
> File <command-1124948972684715>:10
>       8 df = ispark.read_parquet(source = "abfss://my_parquet")
>       9 df = df.select(_.CAMP_START_DATE, _.CAMP_END_DATE, _.KPM_PROJECT_ID)
> ---> 10 ispark.create_table(name = "raw_camp_info", obj = df, overwrite = True, format="delta", database="dart_extensions")
> 
> File /local_disk0/.ephemeral_nfs/envs/pythonEnv-c1f2e765-cb45-4a96-a1dc-bd25ecd4f460/lib/python3.9/site-packages/ibis/backends/pyspark/__init__.py:453, in Backend.create_table(self, name, obj, schema, database, temp, overwrite, format)
>     451         self._run_pre_execute_hooks(table)
>     452         df = self._session.sql(query)
> --> 453         df.write.saveAsTable(name, format=format, mode=mode)
>     454 elif schema is not None:
>     455     schema = PySparkSchema.from_ibis(schema)
> 
> File /usr/lib/python3.9/contextlib.py:124, in _GeneratorContextManager.__exit__(self, type, value, traceback)
>     122 if type is None:
>     123     try:
> --> 124         next(self.gen)
>     125     except StopIteration:
>     126         return False
> 
> File /local_disk0/.ephemeral_nfs/envs/pythonEnv-c1f2e765-cb45-4a96-a1dc-bd25ecd4f460/lib/python3.9/site-packages/ibis/backends/pyspark/__init__.py:234, in Backend._active_database(self, name)
>     232     yield
>     233 finally:
> --> 234     self._session.catalog.setCurrentDatabase(current)
> 
> File /databricks/spark/python/pyspark/instrumentation_utils.py:48, in _wrap_function.<locals>.wrapper(*args, **kwargs)
>      46 start = time.perf_counter()
>      47 try:
> ---> 48     res = func(*args, **kwargs)
>      49     logger.log_success(
>      50         module_name, class_name, function_name, time.perf_counter() - start, signature
>      51     )
>      52     return res
> 
> File /databricks/spark/python/pyspark/sql/catalog.py:185, in Catalog.setCurrentDatabase(self, dbName)
>     172 def setCurrentDatabase(self, dbName: str) -> None:
>     173     """
>     174     Sets the current default database in this session.
>     175 
>    (...)
>     183     >>> spark.catalog.setCurrentDatabase("default")
>     184     """
> --> 185     return self._jcatalog.setCurrentDatabase(dbName)
> 
> File /databricks/spark/python/lib/py4j-0.10.9.5-src.zip/py4j/java_gateway.py:1321, in JavaMember.__call__(self, *args)
>    1315 command = proto.CALL_COMMAND_NAME +\
>    1316     self.command_header +\
>    1317     args_command +\
>    1318     proto.END_COMMAND_PART
>    1320 answer = self.gateway_client.send_command(command)
> -> 1321 return_value = get_return_value(
>    1322     answer, self.gateway_client, self.target_id, self.name)
>    1324 for temp_arg in temp_args:
>    1325     temp_arg._detach()
> 
> File /databricks/spark/python/pyspark/errors/exceptions.py:228, in capture_sql_exception.<locals>.deco(*a, **kw)
>     226 def deco(*a: Any, **kw: Any) -> Any:
>     227     try:
> --> 228         return f(*a, **kw)
>     229     except Py4JJavaError as e:
>     230         converted = convert_exception(e.java_exception)
> 
> File /databricks/spark/python/lib/py4j-0.10.9.5-src.zip/py4j/protocol.py:326, in get_return_value(answer, gateway_client, target_id, name)
>     324 value = OUTPUT_CONVERTER[type](answer[2:], gateway_client)
>     325 if answer[1] == REFERENCE_TYPE:
> --> 326     raise Py4JJavaError(
>     327         "An error occurred while calling {0}{1}{2}.\n".
>     328         format(target_id, ".", name), value)
>     329 else:
>     330     raise Py4JError(
>     331         "An error occurred while calling {0}{1}{2}. Trace:\n{3}\n".
>     332         format(target_id, ".", name, value))
> 
> Py4JJavaError: An error occurred while calling o569.setCurrentDatabase.
> : com.databricks.sql.managedcatalog.acl.UnauthorizedAccessException: PERMISSION_DENIED: User does not have USE SCHEMA on Schema 'comms_media_dev.default'.
> 	at com.databricks.managedcatalog.UCReliableHttpClient.reliablyAndTranslateExceptions(UCReliableHttpClient.scala:84)
> 	at com.databricks.managedcatalog.UCReliableHttpClient.get(UCReliableHttpClient.scala:136)
> 	at com.databricks.managedcatalog.ManagedCatalogClientImpl.$anonfun$getSchema$1(ManagedCatalogClientImpl.scala:378)
> 	at com.databricks.managedcatalog.ManagedCatalogClientImpl.$anonfun$recordAndWrapException$2(ManagedCatalogClientImpl.scala:3401)
> 	at com.databricks.spark.util.FrameProfiler$.record(FrameProfiler.scala:80)
> 	at com.databricks.managedcatalog.ManagedCatalogClientImpl.$anonfun$recordAndWrapException$1(ManagedCatalogClientImpl.scala:3400)
> 	at com.databricks.managedcatalog.ErrorDetailsHandler.wrapServiceException(ErrorDetailsHandler.scala:25)
> 	at com.databricks.managedcatalog.ErrorDetailsHandler.wrapServiceException$(ErrorDetailsHandler.scala:23)
> 	at com.databricks.managedcatalog.ManagedCatalogClientImpl.wrapServiceException(ManagedCatalogClientImpl.scala:106)
> 	at com.databricks.managedcatalog.ManagedCatalogClientImpl.recordAndWrapException(ManagedCatalogClientImpl.scala:3397)
> 	at com.databricks.managedcatalog.ManagedCatalogClientImpl.getSchema(ManagedCatalogClientImpl.scala:374)
> 	at com.databricks.sql.managedcatalog.ManagedCatalogCommon.$anonfun$getSchemaMetadata$5(ManagedCatalogCommon.scala:227)
> 	at scala.Option.getOrElse(Option.scala:189)
> 	at com.databricks.sql.managedcatalog.ManagedCatalogCommon.getSchemaMetadata(ManagedCatalogCommon.scala:227)
> 	at com.databricks.sql.managedcatalog.ManagedCatalogCommon.schemaExists(ManagedCatalogCommon.scala:233)
> 	at com.databricks.sql.managedcatalog.ManagedCatalogSessionCatalog.databaseExists(ManagedCatalogSessionCatalog.scala:631)
> 	at com.databricks.sql.managedcatalog.ManagedCatalogSessionCatalog.requireScExists(ManagedCatalogSessionCatalog.scala:274)
> 	at com.databricks.sql.managedcatalog.ManagedCatalogSessionCatalog.setCurrentDatabase(ManagedCatalogSessionCatalog.scala:492)
> 	at com.databricks.sql.DatabricksCatalogManager.setCurrentNamespace(DatabricksCatalogManager.scala:159)
> 	at org.apache.spark.sql.internal.CatalogImpl.setCurrentDatabase(CatalogImpl.scala:98)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> 	at java.lang.reflect.Method.invoke(Method.java:498)
> 	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
> 	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:380)
> 	at py4j.Gateway.invoke(Gateway.java:306)
> 	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
> 	at py4j.commands.CallCommand.execute(CallCommand.java:79)
> 	at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:195)
> 	at py4j.ClientServerConnection.run(ClientServerConnection.java:115)
> 	at java.lang.Thread.run(Thread.java:750)

If I set the catalog and database via pyspark, `create_table` works as excepted:
```
ispark._session.catalog.setCurrentCatalog("comms_media_dev")
ispark._session.catalog.setCurrentDatabase("dart_extensions")
ispark.create_table(name = "raw_camp_info", obj = df, overwrite = True, format="delta")

**Comment by gforsyth** [comment]
Hey @mark-druffel -- can you open a separate issue for that?  That is definitely an oversight and we can add in `('catalog', 'database')` support for `create_table` but it's helpful to have a separate issue to track it.
